codes/models/modules/architectures/RRDBNet_arch.py

```diff
@@ -65,24 +65,36 @@ class RRDB(nn.Module):
     (ESRGAN: Enhanced Super-Resolution Generative Adversarial Networks)
     '''
 
-    def __init__(self, nf, kernel_size=3, gc=32, stride=1, bias=1, pad_type='zero', \
+    def __init__(self, nf, nr=3, kernel_size=3, gc=32, stride=1, bias=1, pad_type='zero', \
             norm_type=None, act_type='leakyrelu', mode='CNA', convtype='Conv2D', \
             spectral_norm=False, gaussian_noise=False, plus=False):
         super(RRDB, self).__init__()
-        self.RDB1 = ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type, \
-                norm_type, act_type, mode, convtype, spectral_norm=spectral_norm, \
-                gaussian_noise=gaussian_noise, plus=plus)
-        self.RDB2 = ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type, \
-                norm_type, act_type, mode, convtype, spectral_norm=spectral_norm, \
-                gaussian_noise=gaussian_noise, plus=plus)
-        self.RDB3 = ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type, \
-                norm_type, act_type, mode, convtype, spectral_norm=spectral_norm, \
-                gaussian_noise=gaussian_noise, plus=plus)
+        # This is for backwards compatibility with existing models
+        if nr == 3:
+            self.RDB1 = ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type, \
+                    norm_type, act_type, mode, convtype, spectral_norm=spectral_norm, \
+                    gaussian_noise=gaussian_noise, plus=plus)
+            self.RDB2 = ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type, \
+                    norm_type, act_type, mode, convtype, spectral_norm=spectral_norm, \
+                    gaussian_noise=gaussian_noise, plus=plus)
+            self.RDB3 = ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type, \
+                    norm_type, act_type, mode, convtype, spectral_norm=spectral_norm, \
+                    gaussian_noise=gaussian_noise, plus=plus)
+        else:
+            self.RDBs = [ResidualDenseBlock_5C(nf, kernel_size, gc, stride, bias, pad_type,
```

Comment:
Instead of "self.RDBs", it can be "RDBs", and then the list should be passed to nn.Sequential to create the pytorch modules. That way looping through the list in the forward is not needed